src/shell/commands/disk_rebalance.cpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2017, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include "shell/commands.h"
+#include "shell/argh.h"
+
+#include <math.h>
+#include <fmt/ostream.h>
+#include <dsn/utility/errors.h>
+#include <dsn/utility/output_utils.h>
+#include <dsn/utility/string_conv.h>
+#include <dsn/dist/replication/duplication_common.h>
+
+bool query_disk_info(command_executor *e, shell_context *sc, arguments args)
```

Comment:
I think the function is too long. Can you split it into some other functions?

Reply:
Yes, but it because has many `printer` logic.